### DISCUSSION TITLE
Made a go app for querying api endpoints.

### DISCUSSION THREAD

**Discussion by t-e-s-tweb** [0xERR0R/blocky#1047] [Show and tell]
I was looking for a frontend for blocky but the current available ones require too many dependencies for just a simple function. 

So, I made a simple go app which when run requests your domain name, enter the domain name without https:// and the ending / and after that its as simple as choosing options. e.g github.com

<a href="https://imgbb.com/"><img src="https://i.ibb.co/X3J0MVC/1.png" alt="1" border="0"></a>
<a href="https://imgbb.com/"><img src="https://i.ibb.co/S36RhSs/2.png" alt="2" border="0"></a>
<a href="https://imgbb.com/"><img src="https://i.ibb.co/BzQPdHZ/3.png" alt="3" border="0"></a>
<a href="https://imgbb.com/"><img src="https://i.ibb.co/kJFgt05/4.png" alt="4" border="0"></a>

[blockyapiamd64.zip](https://github.com/0xERR0R/blocky/files/11646702/blockyapiamd64.zip)

It would be great if we could read and edit the configuration file and view querylog via api calls. There are some apps like https://github.com/looterz/grimd/blob/master/api.go which have the ability to view detailed query log over api.

**Comment by kwitsch** [comment]
Since the query logs are usually stored in a separate database it would be easier to read them from there instead of reading them via a API endpoint(which would be rather complex implementing SQL features).

Editing the configuration is not possible during the runtime since it would require a restart to activate them. Furthermore is it possible to split the configuration in multiple files or have read-only configurations.
There are quite a few scenarios where writing the configuration back to a file wouldn't work.
